contracts/BookMarketplace.sol

```diff
@@ -44,15 +48,98 @@ contract BookMarketplace is ERC721URIStorage {
         bool sold
     );
 
+    event ListingUpdated(
+        uint256 indexed tokenId,
+        address seller,
+        address owner,
+        uint256 instantPrice,
+        uint256 startingPrice,
+        bool allowBid,
+        Bid[] bidList,
+        bool sold
+    );
+
+    /* modifiers */
+    
+    
+    // Modifier prevents function from being called by anyone other than the market owner
+    modifier onlyMarketOwner() {
+        require(
+            marketOwner == msg.sender,
+            "Only the marketplace owner can call this function."
+        );
+        _;
+    }
+
+    /* constructor */
+
     constructor() ERC721("BookExchange Tokens", "BOOK") {
-        owner = payable(msg.sender);
+        marketOwner = payable(msg.sender);
+    }
+
+    /* functions */
+
+    function getMarketOwnerAddress() public view returns(address) {
+        return marketOwner;
+    }
+
+    // change the price of a listing
+    function updateInstantPrice(uint256 tokenId, uint256 newPrice) public payable onlyMarketOwner {
+        Listing storage listing = idToMarketItem[tokenId];
+        listing.instantPrice = newPrice;
+        idToMarketItem[tokenId] = listing;
+    }
+
+    // get the price of a listing
+    function getInstantPrice(uint256 tokenId) public view returns (uint256) {
+        Listing storage listing = idToMarketItem[tokenId];
+        return listing.instantPrice;
+    }
+
+    // Adds bids to a bid list by tokenId
+    function addBid(uint256 tokenId, uint256 bidAmount) public payable {
```

Comment:
Payable should only be used when we need to transfer fees. If we think bidder has to pay before they can submit bid then payable would be of great use. This is also a discussion @MichaelFauht what do you think?

Reply:
That is what I was thinking, that a user would have to guarantee their bid with a payment, the remainder could be refunded. 

Otherwise bids could fall through and people could artificially ramp up bid price pretty quickly.

Alternatively, is there a way to put a hold on funds in solidity? Like preauthorizing a credit card?

As far as the name goes, I think that both give the right info/impression for usability. I have no preference.

Reply:
Korey makes a really good point. Artificially inflating prices with wash sale-like tactics is an issue in a lot of markets; having to back your bid with funds makes sense to me. Putting a hold on the funds after a bid does make sense, could that cause issues with someone wanting to bid on a separate listing incase one falls through?

Reply:
Korey and I had a discussion on this yesterday and we sort of agreed on introducing a new fee called bidding fees. Essentially, each time (or just the first time) a bidder submit a bid, they'll have to pay a small amount of fees in order to be able to bid. What do you think?

Reply:
I like it. Who would the fees end up being paid out to?

Reply:
Resolving this conversation, see bid commits coming soon.